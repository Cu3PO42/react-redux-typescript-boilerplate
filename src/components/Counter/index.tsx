import React from 'react';

interface CounterProps {
  count: number;
  increment: () => void;
  decrement: () => void;
}

const Counter = ({ count, increment, decrement }: CounterProps) => (
  <div>
    <div>{count}</div>
    <div>
      <button onClick={increment}>+</button>
      <button onClick={decrement}>-</button>
    </div>
  </div>
);

export default Counter;
