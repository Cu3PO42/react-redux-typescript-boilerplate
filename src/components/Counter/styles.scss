.wrapper {
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(to bottom right, #159957, #155799);
  color: white;
  font-size: 2em;
  font-family: 'Fira Mono', 'Operator Mono', 'Courier New', monospace;
  font-weight: 100;

  div:first-child {
    font-size: 1.5em;
  }

  button {
    background-color: transparent;
    border: 2px white solid;
    border-radius: 50%;
    width: 1.5em;
    height: 1.5em;
    color: white;
    font-size: 1.5em;
    vertical-align: center;
    margin: 20px;
    padding: 0;
    font-family: 'Fira Mono', 'Operator Mono', 'Courier New', monospace;
    font-weight: 100;
    outline: none;
    cursor: pointer;
  }
}
