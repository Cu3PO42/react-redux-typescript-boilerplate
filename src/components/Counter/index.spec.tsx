import React from 'react';
import renderer from 'react-test-renderer';
import Counter from '.';

test('Counter is rendered', () => {
  const component = renderer.create(<Counter count={2} increment={() => undefined} decrement={() => undefined} />);
  expect(component.toJSON()).toMatchSnapshot();
});
