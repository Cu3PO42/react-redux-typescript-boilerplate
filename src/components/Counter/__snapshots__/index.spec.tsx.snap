// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter is rendered 1`] = `
<div
  className="wrapper"
>
  <div>
    2
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      +
    </button>
    <button
      onClick={[Function]}
    >
      -
    </button>
  </div>
</div>
`;
